runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -284,7 +290,7 @@ protected boolean doesEntryUpdateContext(final ILogData data) {
      */
     private void updatePointer(final ILogData data) {
         // Update the global pointer, if it is non-checkpoint data.
-        if (data.getType() == DataType.DATA && !data.hasCheckpointMetadata()) {
+        if ((data.isData() || data.isHole()) && !data.hasCheckpointMetadata()) {
```

Comment:
Same here, we might end up pointing to an address belongs to a different stream.

Reply:
Done.